apps/angular/projection/src/app/ui/card/card.component.ts

```diff
@@ -1,61 +1,36 @@
-import { NgFor, NgIf } from '@angular/common';
-import { Component, Input } from '@angular/core';
-import { randStudent, randTeacher } from '../../data-access/fake-http.service';
-import { StudentStore } from '../../data-access/student.store';
-import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
+import { NgStyle, NgTemplateOutlet } from '@angular/common';
+import { Component, ContentChild, Input, TemplateRef } from '@angular/core';
 import { ListItemComponent } from '../list-item/list-item.component';
-
 @Component({
   selector: 'app-card',
   template: `
     <div
       class="flex w-fit flex-col gap-3 rounded-md border-2 border-black p-4"
-      [class]="customClass">
-      <img
-        *ngIf="type === CardType.TEACHER"
-        src="assets/img/teacher.png"
-        width="200px" />
-      <img
-        *ngIf="type === CardType.STUDENT"
-        src="assets/img/student.webp"
-        width="200px" />
-
+      [ngStyle]="{ 'background-color': customClass }">
+      <ng-content select="img"></ng-content>
       <section>
-        <app-list-item
-          *ngFor="let item of list"
-          [name]="item.firstName"
-          [id]="item.id"
-          [type]="type"></app-list-item>
+        @for (item of list; track item.id) {
+          <app-list-item [id]="item.id">
+            <ng-container
+              *ngTemplateOutlet="
+                deleteButton;
+                context: {
+                  $implicit: item.id,
+                  name: item.name,
+                  firstName: item.firstName
```

Comment:
why are you not setting the full object ? 
Because you don't know the property of the inputed list. It can be anything if you want your component to be highly reusable